perf/evp_setpeer.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <libgen.h>
+#include <unistd.h>
+#include <openssl/evp.h>
+#include "perflib/perflib.h"
+
+#define NUM_CALLS_PER_TEST         1000000
+
+int err = 0;
+
+size_t num_calls;
+static int threadcount;
+static OSSL_TIME *times = NULL;
+
+static unsigned char buf[32] = {
```

Comment:
What kind of key is kept in `buf` the task mentions to use more key types:
`It should be able to measure the performance on at least DH ffdhe2048, ECDH P-256, P-521, and X25519.`

the code as-is seems to measure only one particular key type.

I would suggest you to take a look at `pkeyread.c` I feel this new test should look quite similar.